packages/react-server/src/ReactFlightServer.js

```diff
@@ -4363,5 +4363,7 @@ export function abort(request: Request, reason: mixed): void {
   } catch (error) {
     logRecoverableError(request, error, null);
     fatalError(request, error);
+  } finally {
+    callOnAllReadyIfReady(request);
```

Comment:
If we fatal error, we shouldn't also call on onAllReady. In the case of a Promise the native implementation of promises might dedupe the second one (but really that should probably be a warning) but for example the Node.js callback model doesn't do this which would lead to incorrect results for the consumer.

Additionally, if the onAllReady callback itself throws we should trigger the fatal error flow to clean up. So this should be inside the catch.